### PR TITLE
Address UX study feedback and API review feedback

### PR DESCRIPTION
**Pull request by srnagar**
This PR addresses some of the feedback that we received during the user studies and also includes changes that were recommended during the API review.

### CODE REVIEW

File: sdk/monitor/azure-monitor-query/src/main/java/module-info.java [hunk 1]
```diff
@@ -9,9 +9,9 @@
     requires transitive com.azure.core.experimental;
     exports com.azure.monitor.query;
     exports com.azure.monitor.query.models;
-    opens com.azure.monitor.query.log.implementation.models to com.fasterxml.jackson.databind, com.azure.core;
-    opens com.azure.monitor.query.metrics.implementation.models to com.fasterxml.jackson.databind, com.azure.core;
-    opens com.azure.monitor.query.metricsdefinitions.implementation.models to com.fasterxml.jackson.databind, com.azure.core;
-    opens com.azure.monitor.query.metricsnamespaces.implementation.models to com.fasterxml.jackson.databind, com.azure.core;
     opens com.azure.monitor.query.models to com.fasterxml.jackson.databind, com.azure.core;
+    opens com.azure.monitor.query.implementation.logs.models to com.fasterxml.jackson.databind, com.azure.core;
```
Comment (pallavit) [inline sdk/monitor/azure-monitor-query/src/main/java/module-info.java:13]: >.implementation.logs [](http://example.com/codeflow?start=33&length=20)

why the change in the namespace?

Comment (srnagar) [reply sdk/monitor/azure-monitor-query/src/main/java/module-info.java:13]: Jonathan suggested that we have `implementation` as the root and then fork into logs, metrics etc. This follows the pattern we've used in other places.

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/models/LogsQueryHelper.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                                    field.set(t, tableCell.getValueAsDouble());
+                                } else if (tableCell.getColumnType() == LogsColumnType.STRING
+                                        || tableCell.getColumnType() == LogsColumnType.GUID
+                                        || tableCell.getColumnType() == LogsColumnType.TIMESPAN) {
+                                    field.set(t, tableCell.getValueAsString());
+                                }
+                                field.setAccessible(false);
+                            } catch (IllegalAccessException ex) {
+                                throw LOGGER.logExceptionAsError(
+                                        new IllegalArgumentException("Failed to set column value for " + columnName, ex));
```
Comment (pallavit) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/models/LogsQueryHelper.java:142]: Why allow partial creation of the object?

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/models/LogsQueryHelper.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                                    field.set(t, tableCell.getValueAsLong());
+                                } else if (tableCell.getColumnType() == LogsColumnType.REAL
+                                        || tableCell.getColumnType() == LogsColumnType.DECIMAL) {
+                                    field.set(t, tableCell.getValueAsDouble());
+                                } else if (tableCell.getColumnType() == LogsColumnType.STRING
+                                        || tableCell.getColumnType() == LogsColumnType.GUID
+                                        || tableCell.getColumnType() == LogsColumnType.TIMESPAN) {
+                                    field.set(t, tableCell.getValueAsString());
+                                }
+                                field.setAccessible(false);
```
Comment (pallavit) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/models/LogsQueryHelper.java:139]: Should this not be in the finally block to make sure we do not accidently keep a field accessible?

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java [hunk 4]
```diff
@@ -190,25 +239,22 @@ private Context updateContext(Duration serverTimeout, Context context) {
...
             if (errors.getDetails() != null) {
                 errorDetails = errors.getDetails()
                         .stream()
-                        .map(errorDetail -> new LogsQueryErrorDetail(errorDetail.getCode(),
-                                errorDetail.getMessage(),
-                                errorDetail.getTarget(),
-                                errorDetail.getValue(),
-                                errorDetail.getResources(),
-                                errorDetail.getAdditionalProperties()))
+                        .map(errorDetail -> new HttpResponseError(errorDetail.getCode(), errorDetail.getMessage())
```
Comment (pallavit) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java:256]: >errorDetail [](http://example.com/codeflow?start=29&length=11)

Why remove the strongly typed error here and make it generic? Do we not see value in having a new type?

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/MetricsQueryAsyncClient.java [hunk 5]
```diff
@@ -80,10 +95,12 @@
      * @return List of metrics namespaces.
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
+    @SuppressWarnings("deprecation")
```
Comment (pallavit) [inline sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/MetricsQueryAsyncClient.java:98]: Has the swagger deprecated these?

**Review by pallavit — APPROVED** [review]
:shipit:

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/models/LogsQueryHelper.java (hunk 3 above)
Comment (srnagar) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/models/LogsQueryHelper.java:139]: This doesn't permanently make the field accessible. It applies to only the current field instance and setting this back to false is strictly not required since I don't use this field instance anywhere else after this line.

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/models/LogsQueryHelper.java (hunk 2 above)
Comment (srnagar) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/logs/models/LogsQueryHelper.java:142]: This won't create partial objects. If we fail to set the value of any column, this whole method will throw. So, the caller of this method will not get any result.

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/MetricsQueryAsyncClient.java (hunk 5 above)
Comment (srnagar) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/MetricsQueryAsyncClient.java:98]: No, this is not a deprecated method. It is just suppressing warnings since the `mapPage` in azure core is marked as deprecated.

File: sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java (hunk 4 above)
Comment (srnagar) [reply sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java:256]: We have a common error shape defined by the REST API guidelines. All services are expected to use the same error structure. So, we want to define this type once in core and reuse it in all services instead of creating new types in each client library.
